src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -74,17 +74,29 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
                 return;
             }
 
-            // RegisterWorkspaceHost is required to be called from UI thread so push the code
-            // to UI thread to run. 
-            await Task.Factory.SafeStartNew(() =>
+            // Create a connection to the host in the BG to avoid taking the hit of loading service 
+            // hub on the UI thread.  We'll initially set its ref count to 1, and we will decrement 
+            // that ref-count at the end of the using block.  During this time though, when the 
+            // projectTracker is sending events, the workspace host can then use that connection 
+            // instead of having to expensively spin up a fresh one.
+            var currentConnection = await client.TryCreateConnectionAsync(WellKnownRemoteHostServices.RemoteHostService, CancellationToken.None).ConfigureAwait(false);
```

Comment:
how about you just use KeepAliveSession?

http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/SymbolSearch/SymbolSearchUpdateEngineFactory.cs,24

create the session from BG and use TryInvokeAsync